plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt

```diff
@@ -814,7 +831,7 @@ class ClassFileToSourceStubConverter(val kaptContext: KaptContextForStubGenerati
         val evaluator = ConstantExpressionEvaluator(moduleDescriptor, languageVersionSettings, kaptContext.project)
         val trace = DelegatingBindingTrace(kaptContext.bindingContext, "Kapt")
         val const = evaluator.evaluateExpression(expression, trace, expectedType)
-        if (const == null || const.isError || !const.canBeUsedInAnnotations || const.usesNonConstValAsConstant) {
```

Comment:
Is this change intentional? If so, can you please describe why we need this?

Reply:
Yes, it was intentional. That code prevents non annotation compatible values from having an initializer. This means that something like this `val nullable: String? = null` would not get an initializer when I argue that is should.

Also, that method is never called for annotation types so it will not lead to any compile errors for annotations.